File: Java Barcode Scanner ME/app/src/main/java/de/dimedis/mobileentry/model/Ticket.java
package de.dimedis.mobileentry.model;

import androidx.annotation.NonNull;

import com.google.gson.annotations.SerializedName;

public final class Ticket {
    @SerializedName("ticket_code")
    public final String ticketCode;
    @SerializedName("article_id")
    public final String articleId;
    @SerializedName("article_name")
    public final String articleName;

    public Ticket(String ticketCode, String articleId, String articleName) {
        this.ticketCode = ticketCode;
        this.articleId = articleId;
        this.articleName = articleName;
    }

    public static Ticket fromLmLib(@NonNull de.dimedis.lmlib.myinterfaces.Ticket ticket) {
        return new Ticket(ticket.getTicketCode(), ticket.getArticleId(), ticket.getArticleName());
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        Ticket ticket = (Ticket) o;
        return ticketCode.equals(ticket.ticketCode);
    }

    @Override
    public int hashCode() {
        return ticketCode.hashCode();
    }

    @Override
    public String toString() {
        return "Ticket{" + "ticketCode='" + ticketCode + '\'' + ", articleId='" + articleId + '\'' + ", articleName='" + articleName + '\'' + '}';
    }
}

File: Kotlin Migration/NTMobileEntry/app/src/main/java/de/dimedis/mobileentry/model/Ticket.kt
package de.dimedis.mobileentry.model

import com.google.gson.annotations.SerializedName

class Ticket(@field:SerializedName("ticket_code") val ticketCode: String, @field:SerializedName("article_id"
    ) val articleId: String, @field:SerializedName("article_name") val articleName: String) {
    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val ticket = other as Ticket
        return ticketCode == ticket.ticketCode
    }

    override fun hashCode(): Int {
        return ticketCode.hashCode()
    }

    override fun toString(): String {
        return "Ticket{ticketCode='$ticketCode', articleId='$articleId', articleName='$articleName'}"
    }

    companion object {
        fun fromLmLib(ticket: Ticket): Ticket {
            return Ticket(ticket.ticketCode, ticket.articleId, ticket.articleName)
        }
    }
}